### ISSUE TITLE
Modify the HTML before Angular compiles it

### ISSUE THREAD

**Issue by ArielGueta** [just-jeb/angular-builders#989]
**Is your feature request related to a problem? Please describe.**
I want to modify the html before Angular compiles it.

**Describe the solution you'd like**
Hook that will give me the html ( also inline ) before Angular complies it.

**Comment by just-jeb** [comment]
This is totally doable with Webpack & Loaders (not sure though about the inline part).  
There is a bunch of issues with different examples, the solution to all of them is to disable direct template loading. [Here](https://github.com/just-jeb/angular-builders/issues/616) is one of them for example.

**Comment by ArielGueta** [comment]
Thanks. I see we need to change the Angular plugin. Do you think it can be added as hook to this library, like you did with index transform html?

**Comment by just-jeb** [comment]
Actually I did consider it and even consulted with Angular team about it.  

There are two levels of what you propose:
1. Add a flag (like `enableHtmlModification`) that takes care of modifying `directTemplateLoading` in Angular compiler plugin. This would release the users from the burden of looking for this in the config and updating it themselves. 
2. Expose a hook (just like you suggested), which, if provided in options, will add custom html loader AND disable direct template loading.  

Both these options are, however, will affect the very definition of the builder and (as I expect) would dramatically increase the amount of issues and questions.  
Currently the purpose of the builder is simply to provide a way to hook into Webpack build and `index.html` processing. 
The responsibility for the correct config and the code inside of it is on users.  
The moment this builder starts modifying the config itself (not just merging between the Angular and the user config), it will dramatically increase complexity and potential issues root causes.  

Bottom line, at the moment I'd prefer to leave it as is, with its simple purpose.
